OpenAI-DotNet/Assistants/AssistantTool.cs

```diff
@@ -1,12 +1,27 @@
 using System.Text.Json.Serialization;
 using OpenAI.Chat;
+using OpenAI.Extensions;
 
 namespace OpenAI.Assistants
 {
     public sealed class AssistantTool
     {
+        public AssistantTool() { }
+
+        public AssistantTool(AssistantToolType type)
+        {
+            Type = type;
+        }
+
+        public AssistantTool(Function function)
+        {
+            Type = AssistantToolType.Function;
+            Function = function;
+        }
+
         [JsonPropertyName("type")]
-        public string Type { get; set; }
+        [JsonConverter(typeof(JsonStringEnumConverter<AssistantToolType>))]
+        public AssistantToolType Type { get; set; }
```

Comment:
Wondering if we should keep this as a string in case they make it so you can customize it with your own tools

Reply:
> The Assistants API currently supports three types of [tools](https://platform.openai.com/docs/assistants/tools): Code Interpreter, Retrieval, and Function calling. In the future, we plan to release more OpenAI-built tools, and allow you to provide your own tools on our platform.

ok, let's make it string